packages/stack/src/views/Stack/CardContainer.tsx

```diff
@@ -152,8 +165,19 @@ function CardContainer({
     } else {
       onPageChangeCancel?.();
     }
-
-    onTransitionStart?.({ route }, closing);
+    /*fix for https://github.com/react-navigation/react-navigation/issues/9870
+       We trigger transition events only on the incoming (current) screen
+       */
+    if (previousRoute && closing) {
+      //route is closing send to the previous route for navigation.goBack() / navigation.pop / navigation.popToTop()
```

Comment:
The `closing` property is not relevant to the previous route since the previous route isn't being closed, so having a listener fire in a screen that's not transitioning or closing would be confusing.

Reply:
These events are linked to screens animating and not to just the "top" screen. The typical use case is to do something linked to animation starting or ending. this could be on the next / previous screen depending on if the navigation is a pop/push.

 The design of these events is broken as callbacks are removed for useEventEmitter::emit before transitionEnd fires for a closed screen. The changes in this PR do NOT fix the events for closed screens. They do bring the events on par with the corresponding react navigation 3.x events.